### PR TITLE
PopupService,Make parameter meaningful

### PR DESCRIPTION
**Pull request by BaY1251**
### Description of Change ###

 

 ### Proposal ###

 ### PR Checklist ###
 
 - [ ] Has a linked Issue, and the Issue has been `approved`(bug) or `Championed` (feature/proposal)
 - [x] Has tests (if omitted, state reason in description)
 - [ ] Has samples (if omitted, state reason in description)
 - [ ] Rebased on top of `main` at time of PR
 - [ ] Changes adhere to [coding standard](https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#contributing-code---best-practices)
 - [ ] Documentation created or updated: https://github.com/MicrosoftDocs/CommunityToolkit/pulls 

 ### Additional information ###

### CODE REVIEW

File: src/CommunityToolkit.Maui/PopupService.cs [hunk 1]
```diff
@@ -63,7 +63,7 @@ public void ShowPopup<TViewModel>(TViewModel viewModel) where TViewModel : INoti
 
 		var popup = GetPopup(typeof(TViewModel));
 
-		ValidateBindingContext<TViewModel>(popup, out _);
```
Comment (bijington) [inline src/CommunityToolkit.Maui/PopupService.cs:66]: I think we should make sure that the BindingContext is not assigned before assigning it here to make sure we aren't replacing something already created. Mostly to help the developer know if they are making a mistake. What do you think?

**Review by bijington — CHANGES_REQUESTED** [review]

File: src/CommunityToolkit.Maui/PopupService.cs (hunk 1 above)
Comment (BaY1251) [reply src/CommunityToolkit.Maui/PopupService.cs:66]: I agree with you and have modified the code

Comment (Pantheas) [reply src/CommunityToolkit.Maui/PopupService.cs:66]: But isn't setting the `BindingContext` of the page in any case exactly the purpose of this method?
Imagine having a popup where the ViewModel is resolved using DI in most cases.
But for one specific case, we are manually resolving or creating the instance and want to pass it to the popup service.
The instance would never be set as `BindingContext` with this implementation.

Comment (bijington) [reply src/CommunityToolkit.Maui/PopupService.cs:66]: That is correct but I think it best to bring this to the developers attention (potentially with an exception). If they wanted to use your scenario they could potentially register their view models as `AddScoped` and share the instance...

**Comment by brminnick** [comment]
Hi @BaY1251! We plan on deprecating the `public void ShowPopup<TViewModel>(TViewModel)` API. 

I understand your desire, and we've discussed the pros + cons of this in the past in our monthly standups.

However, it's wrong for an API called `ShowPopup` to do any hidden variable assignment under the hood that the caller may be unaware of. It is also a breaking change because it breaks existing functionality for folks currently using the API as designed.

The recommended way to use `PopupService` is to add both the Popup and its ViewModel to `builder.Services` in `MauiProgram.cs`, then call `PopupService.ShowPopup<TViewModel>()`.

**Comment by BaY1251** [comment]
@brminnick This is truly unfortunate news.
I need to use Popup to display the details of list items, so I can only write code like this in the ViewModel 
`Shell. Current.ShowPopup(new MyPopup { BindingContext = viewModel })`.

**Pull request closed without merging**